calicoctl/commands/v1resourceloader/v1resourceloader.go

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1resourceloader
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+
+	log "github.com/sirupsen/logrus"
+
+	yamlsep "github.com/projectcalico/calicoctl/calicoctl/util/yaml"
+	"github.com/projectcalico/go-yaml-wrapper"
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	"github.com/projectcalico/libcalico-go/lib/apis/v1/unversioned"
+	v1validator "github.com/projectcalico/libcalico-go/lib/validator/v1"
+)
+
+// Store a resourceHelper for each resource unversioned.TypeMetadata.
+var resourceToType map[unversioned.TypeMetadata]reflect.Type
+
+func init() {
+	resourceToType = make(map[unversioned.TypeMetadata]reflect.Type)
+	populateResourceTypes()
+}
+
+// populateResourceTypes register all the V1 resource types in the resourceToType map.
+func populateResourceTypes() {
+	resTypes := []unversioned.Resource{
+		apiv1.NewBGPPeer(),
+		apiv1.NewIPPool(),
+		apiv1.NewHostEndpoint(),
+		apiv1.NewNode(),
+		apiv1.NewPolicy(),
+		apiv1.NewProfile(),
+		apiv1.NewWorkloadEndpoint(),
+	}
+
+	for _, rt := range resTypes {
+		resourceToType[rt.GetTypeMetadata()] = reflect.ValueOf(rt).Elem().Type()
+	}
+}
+
+// Create a new concrete resource structure based on the type.  If the type is
+// a list, this creates a concrete Resource-List of the required type.
+func newResource(tm unversioned.TypeMetadata) (unversioned.Resource, error) {
+	rh, ok := resourceToType[tm]
+	if !ok {
+		return nil, errors.New(fmt.Sprintf("Unknown resource type (%s) and/or version (%s)", tm.Kind, tm.APIVersion))
+	}
+	log.Infof("Found resource helper: %s", rh)
```

Comment:
debug level log?

Reply:
hmm I've used `Info` to be consistent with the other calicoctl commands, but I think you're right, it/they should be debug

Reply:
o nvm i thought you were talking about the `log.Infof("results: %+v", results)` line in `convert.go`, but this makes more sense, I'll change it to debug